crypto/s2n_dhe.c

```diff
@@ -180,47 +187,48 @@ int s2n_dh_params_to_p_g_Ys(struct s2n_dh_params *server_dh_params, struct s2n_s
     uint16_t p_size  = BN_num_bytes(bn_p);
     uint16_t g_size  = BN_num_bytes(bn_g);
     uint16_t Ys_size = BN_num_bytes(bn_Ys);
-    uint8_t *p;
-    uint8_t *g;
-    uint8_t *Ys;
+    uint8_t *p = NULL;
+    uint8_t *g = NULL;
+    uint8_t *Ys = NULL;
 
     output->data = s2n_stuffer_raw_write(out, 0);
     notnull_check(output->data);
 
     GUARD(s2n_stuffer_write_uint16(out, p_size));
     p = s2n_stuffer_raw_write(out, p_size);
     notnull_check(p);
-    S2N_ERROR_IF(BN_bn2bin(bn_p, p) != p_size, S2N_ERR_DH_SERIALIZING);
+    ENSURE_POSIX(BN_bn2bin(bn_p, p) == p_size, S2N_ERR_DH_SERIALIZING);
 
     GUARD(s2n_stuffer_write_uint16(out, g_size));
     g = s2n_stuffer_raw_write(out, g_size);
     notnull_check(g);
-    S2N_ERROR_IF(BN_bn2bin(bn_g, g) != g_size, S2N_ERR_DH_SERIALIZING);
+    ENSURE_POSIX(BN_bn2bin(bn_g, g) == g_size, S2N_ERR_DH_SERIALIZING);
 
     GUARD(s2n_stuffer_write_uint16(out, Ys_size));
     Ys = s2n_stuffer_raw_write(out, Ys_size);
     notnull_check(Ys);
-    S2N_ERROR_IF(BN_bn2bin(bn_Ys, Ys) != Ys_size, S2N_ERR_DH_SERIALIZING);
+    ENSURE_POSIX(BN_bn2bin(bn_Ys, Ys) == Ys_size, S2N_ERR_DH_SERIALIZING);
 
     output->size = p_size + 2 + g_size + 2 + Ys_size + 2;
 
-    return 0;
+    return S2N_SUCCESS;
 }
 
 int s2n_dh_compute_shared_secret_as_client(struct s2n_dh_params *server_dh_params, struct s2n_stuffer *Yc_out,
                                            struct s2n_blob *shared_key)
 {
     struct s2n_dh_params client_params = { 0 };
-    uint8_t *            client_pub_key;
-    uint16_t             client_pub_key_size;
-    int                  shared_key_size;
+    uint8_t *            client_pub_key = NULL;
+    uint16_t             client_pub_key_size = 0;
+    int                  shared_key_size = 0;
 
     GUARD(s2n_dh_params_check(server_dh_params));
     GUARD(s2n_dh_params_copy(server_dh_params, &client_params));
     GUARD(s2n_dh_generate_ephemeral_key(&client_params));
     GUARD(s2n_alloc(shared_key, DH_size(server_dh_params->dh)));
 
     const BIGNUM *client_pub_key_bn = s2n_get_Ys_dh_param(&client_params);
+    notnull_check(client_pub_key_bn);
```

Comment:
We could use `ENSURE_REF`/`ENSURE_REF_PTR` macro instead here.

Reply:
Implemented as suggested.